### ISSUE TITLE
Failing tests

### ISSUE THREAD

**Issue by adaugherity** [apereo/mod_auth_cas#157]
I don't know anything about the test framework, and if the tests are even currently in a working state, but I attempted to run them to see if they should be invoked during an RPM build (based on the failures, I'm guessing no).

First of all, I had to fix two build failures of the framework itself:
* In tests/openssl_stubs.c, don't stub out obsolete functions in OpenSSL 1.1 (as I commented about in #155)
* In tests/ap_stubs.c line 68, rename `ap_log_rerror` to `ap_log_rerror_` (added trailing underscore, to match `ap_log_error_` a couple lines above).  This one was confusing since the compile error was in [the Apache headers](http://svn.apache.org/viewvc/httpd/httpd/branches/2.4.x/include/http_log.h?revision=1757669&view=markup#l446):
```
In file included from ap_stubs.c:5:0:
/usr/include/apache2/http_log.h:450:5: error: expected identifier or ‘(’ before ‘do’
     do { if (APLOG_R_MODULE_IS_LEVEL(r, mi, level))                         \
     ^
/usr/include/apache2/http_log.h:448:28: note: in expansion of macro ‘ap_log_rerror__’
 #define ap_log_rerror(...) ap_log_rerror__(__VA_ARGS__)
                            ^~~~~~~~~~~~~~~
ap_stubs.c:68:18: note: in expansion of macro ‘ap_log_rerror’
 AP_DECLARE(void) ap_log_rerror(const char *file, int line, int module_index,
                  ^~~~~~~~~~~~~
/usr/include/apache2/http_log.h:452:7: error: expected identifier or ‘(’ before ‘while’
     } while(0)
       ^
/usr/include/apache2/http_log.h:448:28: note: in expansion of macro ‘ap_log_rerror__’
 #define ap_log_rerror(...) ap_log_rerror__(__VA_ARGS__)
                            ^~~~~~~~~~~~~~~
ap_stubs.c:68:18: note: in expansion of macro ‘ap_log_rerror’
 AP_DECLARE(void) ap_log_rerror(const char *file, int line, int module_index,
                  ^~~~~~~~~~~~~
```
I'm not sure how this ever built with Apache 2.4, since `ap_log_rerror` is a macro and `ap_log_rerror_` the actual function.

Once I got it built, here are the test failures:
> Running suite(s): mod_auth_cas
62%: Checks: 59, Failures: 22, Errors: 0
mod_auth_cas_test.c:236:F:core:getCASScope_test:0: Failed
mod_auth_cas_test.c:524:F:core:setCASCookie_test:0: Assertion 'strcmp(rv, expected) == 0' failed
mod_auth_cas_test.c:538:F:core:setCASCookieExpiryNow_test:0: cookie_name=cookie_value;Path=/; HttpOnly; expires=Thu, 01 Jan 1970 00:00:00 GMT
mod_auth_cas_test.c:550:F:core:setCASCookieExpiryFiveSeconds_test:0: cookie_name=cookie_value;Path=/; HttpOnly; expires=Thu, 01 Jan 1970 00:00:05 GMT
mod_auth_cas_test.c:612:F:core:removeGatewayCookie_test:0: MOD_AUTH_CAS_S=md5;Secure;Path=/; HttpOnly, MOD_CAS_G=TRUE;Secure;Path=/; HttpOnly; expires=Thu, 01 Jan 1970 00:00:00 GMT
mod_auth_cas_test.c:680:F:core:CASCleanCache_test:0: Failed
mod_auth_cas_test.c:728:F:core:createCASCookie_test:0: Failed
mod_auth_cas_test.c:738:F:core:expireCASST_test:0: Failed
mod_auth_cas_test.c:743:F:core:CASSAMLLogout_test:0: Failed
mod_auth_cas_test.c:748:F:core:deleteCASCacheFile_test:0: Failed
mod_auth_cas_test.c:959:F:core:isValidCASCookie_test:0: Failed
mod_auth_cas_test.c:975:F:core:cas_curl_ssl_ctx_test:0: Failed
mod_auth_cas_test.c:1009:F:core:cas_authenticate_test:0: Failed
mod_auth_cas_test.c:1014:F:core:cas_ssl_locking_callback_test:0: Failed
mod_auth_cas_test.c:1029:F:core:cas_cleanup_test:0: Failed
mod_auth_cas_test.c:1034:F:core:check_vhost_config_test:0: Failed
mod_auth_cas_test.c:1039:F:core:check_merged_vhost_configs_test:0: Failed
mod_auth_cas_test.c:1044:F:core:merged_vhost_configs_exist_test:0: Failed
mod_auth_cas_test.c:1049:F:core:cas_post_config_test:0: Failed
mod_auth_cas_test.c:1054:F:core:cas_in_filter_test:0: Failed
mod_auth_cas_test.c:1059:F:core:cas_register_hooks_test:0: Failed
mod_auth_cas_test.c:1263:F:core:cas_attribute_authz_test:0: Assertion '(should_fail1 == AUTHZ_DENIED) && (should_fail2 == AUTHZ_DENIED) && (should_fail3 == AUTHZ_DENIED) && (should_fail4 == AUTHZ_DENIED) && (should_fail5 == AUTHZ_DENIED) && (should_fail6 == AUTHZ_DENIED) && (should_succeed1 == AUTHZ_GRANTED) && (should_succeed2 == AUTHZ_GRANTED) && (should_succeed3 == AUTHZ_GRANTED)' failed
FAIL check_mac (exit status: 1)

This is on openSUSE Leap 15.0, gcc 7.3.1, apache2-2.4.33-lp150.2.3.1.x86_64, openssl-1_1-1.1.0h-lp150.3.6.1.x86_64.

**Comment by dhawes** [comment]
I had to make the same change today to get the tests to build. I get similar failures.

I went ahead and pushed the ap_stubs.c fix to #155.

It goes without saying that the tests need a lot of attention.

**Comment by dhawes** [comment]
Tests still need a lot of work. Closing this for now.
